frontend/svelte/src/lib/modals/neurons/AddHotkeyModal.svelte

```
@@ -13,23 +13,46 @@
   export let neuronId: NeuronId;
 
   let address: string = "";
-  let showAddressNotValid: boolean = false;
+  let validPrincipal: Principal | undefined;
+  $: {
+    try {
+      validPrincipal = Principal.fromText(address);
+    } catch (_) {
+      validPrincipal = undefined;
+    }
+  }
+  let showError: boolean = false;
   let loading: boolean = false;
 
-  const dispatcher = createEventDispatcher();
-  const add = async () => {
-    let principal: Principal | undefined = undefined;
+  // Used to hide error when address changes.
+  let prevValidatedAddress: string | undefined;
```

do we really want to not hide the error when user type?

to some extension we hide the error but then on blur the error can popup again, so is it really useful?

should we validate the adress on input with a debounce aswell?

---

I followed what Mischa [wrote](https://www.notion.so/Form-validation-messages-on-dark-background-43afe1a80b0b402c945855b5cdd9eb16#c0ffa14328fb4579a9ef26ac0bf3faec):

> - Validate on blur (add validation message/error on blur)
> - Remove validation message/error on change

---

Would just `$: value, showError = false` work then?

---

It does! I love these Svelte tricks, nice one!